opentelemetry-java-instrumentation.yaml

```diff
@@ -0,0 +1,60 @@
+package:
+  name: opentelemetry-java-instrumentation
+  version: "2.16.0"
+  epoch: 0
+  description: OpenTelemetry auto-instrumentation and instrumentation libraries for Java
+  copyright:
+    - license: Apache-2.0
+  dependencies:
+    runtime:
+      - busybox
+      # upstream simply adds the platform independent jar to the destdir
+      # https://github.com/open-telemetry/opentelemetry-operator/blob/main/autoinstrumentation/java/Dockerfile
+      # when we build the agent on aarch64 it errors out as some dependencies are not available for aarch64
+      # Cannot find a Java installation on your machine (Linux 6.14.7-r0-linux aarch64) matching: {languageVersion=11, vendor=any vendor, implementation=vendor-specific, nativeImageCapable=false}. Some toolchain resolvers had internal failures: foojay (Service 'SystemInfo' is not available (os=Linux 6.14.7-r0-linux aarch64, enabled=false).).
+      - openjdk-21-default-jdk
+
+environment:
+  contents:
+    packages:
+      - busybox
+      - libudev
+      - openjdk-11
+      - openjdk-17
+      - openjdk-21
+  environment:
+    JAVA_HOME: /usr/lib/jvm/java-21-openjdk
```

Comment:
Let's have a variable above called `java-version` and replace `21` to `${{vars.java-version}}` everywhere.